homeassistant/components/vrm_forecasts/sensor.py

```diff
@@ -0,0 +1,248 @@
+        key="energy_consumption_estimate_tomorrow",
+        translation_key="energy_consumption_estimate_tomorrow",
+        state=lambda estimate: estimate.consumption.tomorrow_total,
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+        suggested_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        suggested_display_precision=1,
+    ),
+    VRMForecastsSensorEntityDescription(
+        key="consumption_highest_peak_time_yesterday",
+        translation_key="consumption_highest_peak_time_yesterday",
+        state=lambda estimate: estimate.consumption.yesterday_peak_time,
+        device_class=SensorDeviceClass.TIMESTAMP,
+    ),
+    VRMForecastsSensorEntityDescription(
+        key="consumption_highest_peak_time_today",
+        translation_key="consumption_highest_peak_time_today",
+        state=lambda estimate: estimate.consumption.today_peak_time,
+        device_class=SensorDeviceClass.TIMESTAMP,
+    ),
+    VRMForecastsSensorEntityDescription(
+        key="consumption_highest_peak_time_tomorrow",
+        translation_key="consumption_highest_peak_time_tomorrow",
+        state=lambda estimate: estimate.consumption.tomorrow_peak_time,
+        device_class=SensorDeviceClass.TIMESTAMP,
+    ),
+    VRMForecastsSensorEntityDescription(
+        key="energy_consumption_current_hour",
+        translation_key="energy_consumption_current_hour",
+        state=lambda estimate: estimate.consumption.current_hour_total,
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+        suggested_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        suggested_display_precision=1,
+    ),
+    VRMForecastsSensorEntityDescription(
+        key="energy_consumption_next_hour",
+        translation_key="energy_consumption_next_hour",
+        state=lambda estimate: estimate.consumption.next_hour_total,
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.WATT_HOUR,
+        suggested_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        suggested_display_precision=1,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: VRMForecastsConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
+) -> None:
+    """Defer sensor setup to the shared sensor module."""
+    coordinator = entry.runtime_data
+
+    async_add_entities(
+        VRMForecastsSensorEntity(
+            entry_id=entry.entry_id,
+            coordinator=coordinator,
+            entity_description=entity_description,
+        )
+        for entity_description in SENSORS
+    )
+
+
+class VRMForecastsSensorEntity(
+    CoordinatorEntity[VRMForecastsDataUpdateCoordinator], SensorEntity
+):
+    """Defines a VRM Solar Forecast sensor."""
+
+    entity_description: VRMForecastsSensorEntityDescription
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        *,
```

Comment:
Personally not a fan of kwargs if all the parameters are the same names as the variables